extensions/slack/src/search-messages.tsx

```diff
@@ -0,0 +1,125 @@
+import { getSlackWebClient } from "./shared/client/WebClient";
+import { convertTimestampToDate, handleError } from "./shared/utils";
+import { useChannels, useUsers } from "./shared/client";
+import { SearchMessagesArguments } from "@slack/web-api";
+
+function Search() {
+  const [query, setQuery] = useState("");
+  const [selectedChannel, setSelectedChannel] = useState("");
+  const [sortType, setSortType] = useCachedState<SearchMessagesArguments["sort"]>("search-sort-type", "timestamp");
+
+  const { data: users } = useUsers();
+  const { data: channels } = useChannels();
+
+  const { data, isLoading } = useCachedPromise(
+    async (query, channel, sort) => {
+      const webClient = getSlackWebClient();
+      const results = await webClient.search.messages({
+        query: `${query}${selectedChannel ? ` in:${channel}` : ""}`,
+        sort,
+      });
+      return results.messages?.matches ?? [];
+    },
+    [query, selectedChannel, sortType],
+    {
+      onError(error) {
+        handleError(error);
+      },
+      execute: query.length > 0,
+      keepPreviousData: true,
+    },
+  );
+
+  const sortOptions: { title: string; sort: SearchMessagesArguments["sort"] }[] = [
+    { title: "Newest", sort: "timestamp" },
+    { title: "Most Relevant", sort: "score" },
+  ];
+
+  const FromMeOnlyAction = () => (
+    <Action
+      title="From Me Only"
+      icon={Icon.Person}
+      shortcut={{ modifiers: ["cmd", "shift"], key: "i" }}
+      onAction={() => setQuery(query ? `from:me ${query}` : "from:me ")}
+    />
+  );
+
+  return (
+    <List
+      isLoading={isLoading}
+      searchText={query}
+      onSearchTextChange={setQuery}
+      actions={
+        <ActionPanel>
+          <FromMeOnlyAction />
+        </ActionPanel>
+      }
+      throttle
+      isShowingDetail
+      searchBarAccessory={
+        <List.Dropdown tooltip="Search Channels" onChange={setSelectedChannel} storeValue>
+          <List.Dropdown.Item value="" title="All Channels" />
+
+          <List.Dropdown.Section>
+            {channels?.flat().map((c) => {
+              return <List.Dropdown.Item key={c.id} icon={c.icon} value={c.name} title={c.name} />;
+            })}
+          </List.Dropdown.Section>
+        </List.Dropdown>
+      }
+    >
+      <List.EmptyView title={query.length > 0 && !isLoading ? "No messages found" : "Search for Slack Messages"} />
+      {data?.map((m) => {
+        if (!m.text || !m.ts) return null;
+
+        const user = users?.find((u) => u.id === m.user);
+        const date = convertTimestampToDate(m.ts);
+        const text = emoji.emojify(m.text);
+
+        return (
+          <List.Item
+            key={m.iid}
+            icon={{ value: user?.icon, tooltip: user?.name ?? "Unknown user" }}
+            title={text}
+            accessories={[{ date, tooltip: format(date, "EEEE dd MMMM yyyy 'at' HH:mm") }]}
+            detail={<List.Item.Detail markdown={text} />}
```

Comment:
question (non-blocking): should we add extra metadata to disambiguate messages from different channels when the search query is just `from:me`?

Reply:
I'm not sure I got what you mean here? Which metadata should we add?

Reply:
Oh, I see. Yes, we could use some metadata; the channel name is a great example.